### DISCUSSION TITLE
named_timezones extension with subsecond precision

### DISCUSSION THREAD

**Discussion by davekaro** [jeremyevans/sequel#1943] [General]
Once I started using the named_timezones extension, I noticed the created_at/updated_at timestamps lost all the subsecond precision. This occurs when setting the `Sequel.application_timestamp` to something other than `:utc` (or presumably `:local` too).

I traced it to happen at this point https://github.com/jeremyevans/sequel/blob/90f6fe5a677333fde49cd31939a84152988f0790/lib/sequel/extensions/named_timezones.rb#L85 where a new `Time` object is created only using the seconds of the original. 

Looking at https://rubyapi.org/3.1/o/time#method-c-at, you can pass the nanoseconds as well (all the way back to Ruby 2.6). Is there a reason not to include the nanoseconds of the original here? I created another extension with this method modified:

```
def convert_output_time_other(v, output_timezone)
  Time.at(v.to_i, v.nsec, :nsec, in: output_timezone)
end
```

and it's working great.

**Comment by jeremyevans** [comment]
Thanks for the heads up.  I'll try to get this supported added before the next release.

**Reply by jeremyevans** [reply]
Fixed at 850c9f90fb41095a10d49d610e70edcf936d9677
